internal/provider/group_resource.go

```diff
@@ -0,0 +1,366 @@
+}
+
+// groupResource is the resource implementation.
+type groupResource struct {
+    client ProviderHTTPClient
+}
+
+// Metadata returns the resource type name.
+func (r *groupResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+    resp.TypeName = req.ProviderTypeName + "_group"
+}
+
+// Schema defines the schema for the group resource.
+func (r *groupResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.Int64Attribute{
+				Required: true,
+			},
+                        "inventory_id": schema.Int64Attribute{
+				Optional: true,
+			},
+			"name": schema.StringAttribute{
+				Required: true,
+			},
+                        "description": schema.StringAttribute{
+                                Optional: true,
+                                Computed:    true,
+                        },
+                        "group_url": schema.StringAttribute{
+				Computed: true,
+			},
+                        "variables": schema.StringAttribute{
+				Optional:   true,
+                                CustomType: jsontypes.NormalizedType{},
+			},
+		},
+	}
+}
+
+// groupResourceModel maps the resource schema data.
+type groupResourceModel struct {
+	Id    types.Int64  `tfsdk:"id"`
+        InventoryId   types.Int64  `tfsdk:"inventory_id"`
+	Name          types.String `tfsdk:"name"`
+	Description          types.String `tfsdk:"description"`
+        URL           types.String         `tfsdk:"group_url"`
+        Variables jsontypes.Normalized `tfsdk:"variables"`
+}
+
+func IsValueProvided(value attr.Value) bool {
+	return !value.IsNull() && !value.IsUnknown()
+}
+
+func (d *groupResourceModel) GetURL() string {
+        if !d.URL.IsNull() && !d.URL.IsUnknown() {
+		return d.URL.ValueString()
+	}
+	return ""
+}
+
+func GetFunctionName(caller int) string {
+        pc, _, _, result := runtime.Caller(caller + 1)
+        if !result {
+                return ""
+        }
+        f := runtime.FuncForPC(pc)
+        if f == nil {
+                return ""
+        }
+        return f.Name()
+}
+
+func (d *groupResourceModel) CreateRequestBody() (*bytes.Reader, diag.Diagnostics) {
+	body := make(map[string]interface{})
+        var diags diag.Diagnostics
+
+	// Inventory id
+        body["inventory"] = d.InventoryId.ValueInt64()
+
+	// Name
+	body["name"] = d.Name.ValueString()
+
+        // Variables
+        if IsValueProvided(d.Variables) {
+                // vars := make(map[string]interface{})
+                // vars_data := []byte(d.Variables)
+                // var replacer = strings.NewReplacer("\r", "", "\n", "")
+                // vars_data := replacer.Replace(d.Variables)
+                // diags.Append(d.Variables.Unmarshal(&vars)...)
+		// if diags.HasError() {
+		//	return nil, diags
+		// }
+		body["variables"] = d.Variables.ValueString()
```

Comment:
I think this needs to be unmarshalled, otherwise when you marshal the body later it will end up as just an escaped json string.

Reply:
Upon unmarshalling, the request is not accepted. As far as I understand, the expected format for variables is a string.